packages/block-editor/src/components/media-replace-flow/style.scss

```diff
@@ -44,6 +44,14 @@
 			padding: 0;
 		}
 
+		.block-editor-link-control__search-item-icon {
+			top: 1em;
+		}
```

Comment:
<img width="919" alt="Screen Shot 2021-07-30 at 09 28 06" src="https://user-images.githubusercontent.com/444434/127624981-5bb758c7-6ec7-4e50-86a3-384796e79403.png">

Not sure this is necessary any longer as I was unable to replicate. I merged a PR recently which sorted out all this globe icon alignment so hopefully that's done it.